index.js

```diff
@@ -112,34 +109,24 @@ class Provider {
     this.id = `${config.name}_${sha}`
 
     const { memopts } = this.args
-    this.fetch = this._makeCached(this.fetch, memopts)
-    this.detail = this._makeCached(this.detail, memopts)
+    this.fetch = this._makeCached(this.fetch.bind(this), memopts)
+    this.detail = this._makeCached(this.detail.bind(this), memopts)
   }
 
-  _makeCached(method, memopts) {
-    const self = this
+  _makeCached (method, memopts) {
     const memoizedMethod = memoize(method, memopts)
 
-    return function () {
-      // XXX: Should be replaced with spread operator if possible.
-      return memoizedMethod.apply(this, arguments)
+    return (...args) => {
+      return memoizedMethod(...args)
         .catch(err => {
           // Delete the cached result if we get an error so retry will work
-          memoizedMethod.delete(self.filters)
+          memoizedMethod.delete(this.filters)
           return Promise.reject(err)
         })
     }
   }
 
-  _parseArgs(argString) {
-    return Provider.parseArgs(argString, this.config.argTypes)
```

Comment:
_parseArgs and parseArgs are not equivalent, one gets argTypes from config

Reply:
Yet it is a private method which is never used inside the `Provider` class, so this just leaves me confused now.

Reply:
it's used in tests

Reply:
it ... is used in the class.